src/checkout-buttons/strategies/masterpass-button-strategy.ts

```diff
@@ -0,0 +1,110 @@
+import { Checkout, CheckoutActionCreator, CheckoutStore } from '../../checkout';
+import {
+    InvalidArgumentError,
+    MissingDataError,
+    MissingDataErrorType
+} from '../../common/error/errors';
+import { PaymentMethod, PaymentMethodActionCreator } from '../../payment';
+import { MasterpassScriptLoader } from '../../payment/strategies/masterpass';
+import { CheckoutButtonInitializeOptions, CheckoutButtonOptions } from '../checkout-button-options';
+
+import CheckoutButtonStrategy from './checkout-button-strategy';
+
+export default class MasterpassButtonStrategy extends CheckoutButtonStrategy {
+    private _signInButton?: HTMLElement;
+    private _paymentMethod?: PaymentMethod;
+    private _checkout?: Checkout;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _checkoutActionCreator: CheckoutActionCreator,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _masterpassScriptLoader: MasterpassScriptLoader
+    ) {
+        super();
+    }
+
+    initialize(options: CheckoutButtonInitializeOptions): Promise<void> {
+        const { masterpass: masterpassOptions, methodId } = options;
+
+        if (!masterpassOptions || !methodId) {
+            throw new InvalidArgumentError('Unable to proceed because "options.masterpass" argument is not provided.');
+        }
+
+        const states = Promise.all(
+            [
+                this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId)),
+                this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout()),
+            ]
+        );
+
+        return states.then(state => {
+                const statePaymentMethod = state[0];
+                const stateCheckout = state[1];
+
+                if (this._isInitialized) {
```

Comment:
This should be the first thing to check in the initialize, so you don't do unnecessary calls if this is true.